### PR TITLE
AP_RangeFinder: Add TeraRanger Neo Rangefinder

### PR DESCRIPTION
**Pull request by Hwurzburg**
New rangefinder....sample will be sent to me for testing in early July...@tridge will share spec directly to you...would appreciate a cursory review since this is my first driver....thanks

### CODE REVIEW

**Comment by Hwurzburg** [comment]
Works perfectly on the bench....will flight test for autoland next

File: libraries/AP_RangeFinder/AP_RangeFinder.cpp [hunk 1]
```diff
@@ -537,6 +538,11 @@ void RangeFinder::detect_instance(uint8_t instance, uint8_t& serial_instance)
     case Type::BenewakeTF03:
 #if AP_RANGEFINDER_BENEWAKE_TF03_ENABLED
         serial_create_fn = AP_RangeFinder_Benewake_TF03::create;
+#endif
+        break;
+#if AP_RANGEFINDER_TERARANGER_SERIAL_ENABLED
```
Comment (tridge) [inline libraries/AP_RangeFinder/AP_RangeFinder.cpp:543]: the #if needs to be one line down

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_RangeFinder/AP_RangeFinder.cpp (hunk 1 above)
Comment (Hwurzburg) [reply libraries/AP_RangeFinder/AP_RangeFinder.cpp]: done

**Comment by Hwurzburg** [comment]
@tridge ready to merge....flight tested....matches terrain height very well...leads baro alt a bit as expected ....height clamp works as expected....while it can measure the 30m max, its really only reliable to 22m, which is still pretty good...

Screenshot from 2022-08-04 10-29-58
Screenshot from 2022-08-04 10-31-03

**Review by tridge — APPROVED** [review]

**Pull request merged**